cmake/HPX_SetupBoost.cmake

```diff
@@ -138,7 +140,7 @@ endif()
 
 include_directories(SYSTEM ${Boost_INCLUDE_DIRS})
 link_directories(${Boost_LIBRARY_DIRS})
-if(NOT MSVC)
+if((NOT MSVC) OR HPX_WITH_VCPKG)
```

Comment:
Sure, this if is only related to wether include the boost libraries explicitly or not. It appears to me, that this is summed up in the `HPX_WITH_BOOST_ALL_DYNAMIC_LINK` option, which might be set independently of `HPX_WITH_VCPKG`. Everything else seems to be fine.